src/esp/physics/PhysicsManager.h

```diff
@@ -105,6 +105,44 @@ struct ContactPointData {
   ESP_SMART_POINTERS(ContactPointData)
 };
 
+//! describes the type of a rigid constraint.
+enum class RigidConstraintType {
+  //! lock a point in one frame to a point in another with no orientation
+  //! constraint
+  PointToPoint,
+  //! fix one frame to another constraining relative position and orientation
+  Fixed
+};
+
+/**
+ * @brief Stores rigid constraint parameters for creation and updates.
+ */
+struct RigidConstraintSettings {
+ public:
+  RigidConstraintSettings() = default;
+
+  //! The type of constraint described by these settings. Determines which
+  //! parameters to use for creation and update.
+  RigidConstraintType constraintType = RigidConstraintType::PointToPoint;
+
+  //! The maximum impulse applied by this constraint. Should be tuned relative
+  //! to physics timestep.
+  double maxImpulse = 1000.0;
+
+  //! object and link ids. objectIdB == ID_UNDEFINED indicates "world".
+  //! objectIdA must always be >= 0.
+  int objectIdA, objectIdB, linkIdA, linkIdB = ID_UNDEFINED;
```

Comment:
Not sure this does what you think it does. Only the last value will be set to ID_UNDEFINED.